docs/kratos/social-signin/35_gitlab.mdx

```diff
@@ -4,8 +4,87 @@ title: Add GitLab as a social sign-in provider in Ory
 sidebar_label: GitLab
 ---
 
+```mdx-code-block
+import JsonnetWarning from "../../_common/jsonnetwarning.mdx"
+```
+
 # GitLab
 
+````mdx-code-block
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+<Tabs>
+<TabItem value="console" label="Ory Console" default>
+
+Follow these steps to add GitLab as a social sign-in provider to your project using the Ory Console:
+
+1. Sign in to Ory Console and select [**Social Sign-in**](https://console.ory.sh/projects/current/social-signin).
+2. Click the switch next to the GitLab logo to start the configuration.
+3. Copy the Redirect URI and save it for later use.
+4. [Create a GitLab OAuth2 Application](https://docs.gitlab.com/ee/integration/oauth_provider.html#adding-an-application-through-the-profile).
+5. Paste the saved **Redirect URI** from Ory into the **Redirect URI** field in the GitLab form.
+6. Enable the relevant **Scopes** for the application, by ticking the relevant boxes in the GitLab form:
+    - `read_user`
+    - `openid`
+    - `profile`
+    - `email`
+7. After clicking **Save application** on the GitLab form, you see a summary of the registered application's properties.
+
+    :::caution
+
+    Make sure to copy the **Secret** field on this summary page right away.
+    This is the only opportunity you will get to copy the secret.
+
+    :::
+
+8. Copy the **Secret** from the registered application in GitLab to the **Client Secret** field on the Ory Console.
+9. Copy the **Application ID** from the registered application in GitLab to the **Client ID** field in the form on the Ory Console.
+10. In the **Scopes** field of the form on the Ory Console, add the following scopes (one by one):
+    - `read_user`
+    - `openid`
+    - `profile`
+    - `email`
+11. In the **Data Mapping** field of the form on the Ory Console, add the following Jsonnet code snippet:
+
+   ```jsonnet
+   local claims = {
+     email_verified: false,
+   } + std.extVar('claims');
+   {
+     identity: {
+       traits: {
+         // Allowing unverified email addresses enables account
+         // enumeration attacks,  if the value is used for
+         // verification or as a password login identifier.
+         //
+         // Therefore we only return the email if it (a) exists and (b) is marked verified
+         // by GitLab.
+         [if 'email' in claims && claims.email_verified then 'email' else null]: claims.email,
+       },
+     },
+   }
+   ```
+
+   :::info
+
+   [GitLab](https://docs.gitlab.com/ee/integration/openid_connect_provider.html) returns only the `sub` and `sub_legacy` claims in
+   the `id_token`. Ory makes a request to [GitLab's /oauth/userinfo API](https://gitlab.com/oauth/userinfo) and adds the user info
+   to `std.extVar('claims')`.
+
+   :::
+
+   ```mdx-code-block
+   <JsonnetWarning format="Jsonnet code snippets" use="data mapping" />
+   ```
+
+12. Click **Save Configuration**.
+
+
+
+</TabItem>
+<TabItem value="cli" label="Ory CLI">
```

Comment:
In the CLI tab you can adjust the indent for the jsonnet admonition: 
<img width="720" alt="image" src="https://user-images.githubusercontent.com/29116492/214261019-8796c5ee-51a9-4de0-8d13-38896eba3139.png">

Reply:
It seems you haven't really touched this for any provider. Please make sure you adjust this. Thanks.

Reply:
Adjusted the indents of the JSonnet admonitions.